### PR TITLE
[SYCL-MLIR] Change sycl.constructor assembly format

### PR DESCRIPTION
**Pull request by victor-eds**
Define new assembly format closer to a func.call:

`$TypeName ( $Args ) attr-dict : ( type($Args) )`

Signed-off-by: Victor Perez <victor.perez@codeplay.com>

### CODE REVIEW

File: mlir-sycl/lib/Dialect/IR/SYCLOps.cpp [hunk 1]
```diff
@@ -67,6 +67,9 @@ bool SYCLCastOp::areCastCompatible(TypeRange Inputs, TypeRange Outputs) {
 }
 
 LogicalResult SYCLConstructorOp::verify() {
+  if (getNumOperands() == 0)
+    return emitOpError("A sycl::constructor must be passed the object to build "
+                       "as the first argument");
```
Comment (victor-eds) [inline mlir-sycl/lib/Dialect/IR/SYCLOps.cpp:72]: We cannot test this using lit tests unfortunately.

File: mlir-sycl/test/Dialect/IR/SYCL/constructor.mlir [hunk 2]
```diff
@@ -1,35 +0,0 @@
-// RUN: sycl-mlir-opt -allow-unregistered-dialect %s -split-input-file | FileCheck %s
```
Comment (victor-eds) [inline mlir-sycl/test/Dialect/IR/SYCL/constructor.mlir:1]: We will store all tests in `ops.mlir`

Comment (victor-eds) [reply mlir-sycl/test/Dialect/IR/SYCL/constructor.mlir:1]: Oh, I see what you mean now. I don't think it's worth it testing that here. Our definition/verification is not extensive enough anyway.

**Review by etiotto — COMMENTED** [review]
LGTM

**Comment by etiotto** [comment]
@victor-eds some of the tests failed to run correctly (/ L0 GEN9 LLVM Test Suite) and (OCL GEN9 LLVM Test Suite) is not complete. I think in this case the PR is not going to affect any tests but in general we should wait that for the LLVM test suite runs to complete and be clean before merging a PR.

**Comment by victor-eds** [comment]
> @victor-eds some of the tests failed to run correctly (L0 GEN9 LLVM Test Suite) and OCL GEN9 LLVM Test Suite is not done running. I think in this case the PR is not going to affect any tests but in general we should wait that for the LLVM test suite runs to complete and be clean before merging a PR.

These tests were failing before. Also, as we only change text representation here and verify we're not creating plain wrong constructors (no argument passed, not possible through `cgeist`), I thought we could just merge.

**Pull request merged**